<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>qSlicerICPRegistrationModuleWidget</class>
 <widget class="qSlicerWidget" name="qSlicerICPRegistrationModuleWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>597</width>
    <height>583</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="ctkCollapsibleButton" name="CollapsibleButton">
     <property name="text">
      <string>Inputs</string>
     </property>
     <property name="contentsFrameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QGroupBox" name="landmarkGroupBox">
        <property name="title">
         <string>Landmarks</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Landmark Transform Mode:</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QRadioButton" name="rigidRadioButton">
           <property name="text">
            <string>Rigid Body</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="similarRadioButton">
           <property name="text">
            <string>Similar</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="affineRadioButton">
           <property name="text">
            <string>Affine</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="meanGroupBox">
        <property name="title">
         <string>Distance Measure</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Mean Distance Mode:</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QRadioButton" name="rmsRadioButton">
           <property name="text">
            <string>RMS</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="absRadioButton">
           <property name="text">
            <string>Absolute Value</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="dataGroupBox">
        <property name="title">
         <string>Registration Parameters</string>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="2" column="1">
          <widget class="QCheckBox" name="centroidsCheckBox">
           <property name="text">
            <string>Start by matching centroids</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Max Number of Iterations:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <spacer name="horizontalSpacer_4">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="0" column="2">
          <widget class="QSpinBox" name="iterationsSpinBox">
           <property name="minimum">
            <number>10</number>
           </property>
           <property name="maximum">
            <number>1000</number>
           </property>
           <property name="singleStep">
            <number>10</number>
           </property>
           <property name="value">
            <number>50</number>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QSpinBox" name="landmarksSpinBox">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>2000</number>
           </property>
           <property name="singleStep">
            <number>10</number>
           </property>
           <property name="value">
            <number>200</number>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QCheckBox" name="meanDistanceCheckBox">
           <property name="text">
            <string>Check mean distance</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Max Number of Landmarks:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <spacer name="horizontalSpacer_3">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="4" column="1">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>Maximum mean distance</string>
           </property>
          </widget>
         </item>
         <item row="4" column="2">
          <widget class="QDoubleSpinBox" name="meanDistanceDoubleSpinBox">
           <property name="decimals">
            <number>3</number>
           </property>
           <property name="minimum">
            <double>0.001000000000000</double>
           </property>
           <property name="maximum">
            <double>10.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.010000000000000</double>
           </property>
           <property name="value">
            <double>0.010000000000000</double>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <spacer name="horizontalSpacer_5">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string>IO</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="2" column="1">
          <widget class="qMRMLNodeComboBox" name="inputMRMLNodeComboBox">
           <property name="nodeTypes">
            <stringlist>
             <string>vtkMRMLModelNode</string>
             <string></string>
            </stringlist>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>Input Surface:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_8">
           <property name="text">
            <string>Target Surface:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>Generated Transform:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="qMRMLNodeComboBox" name="targetMRMLNodeComboBox">
           <property name="nodeTypes">
            <stringlist>
             <string>vtkMRMLModelNode</string>
            </stringlist>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="qMRMLNodeComboBox" name="outputMRMLNodeComboBox">
           <property name="nodeTypes">
            <stringlist>
             <string>vtkMRMLLinearTransformNode</string>
            </stringlist>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="registrationButton">
        <property name="text">
         <string>Register Models</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>qMRMLNodeComboBox</class>
   <extends>QWidget</extends>
   <header>qMRMLNodeComboBox.h</header>
   <slots>
    <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
    <slot>setMRMLScene(vtkMRMLScene*)</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>qSlicerWidget</class>
   <extends>QWidget</extends>
   <header>qSlicerWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkCollapsibleButton</class>
   <extends>QWidget</extends>
   <header>ctkCollapsibleButton.h</header>
   <container>1</container>
   <slots>
    <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>qSlicerICPRegistrationModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>inputMRMLNodeComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>298</x>
     <y>291</y>
    </hint>
    <hint type="destinationlabel">
     <x>351</x>
     <y>414</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerICPRegistrationModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>targetMRMLNodeComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>298</x>
     <y>291</y>
    </hint>
    <hint type="destinationlabel">
     <x>351</x>
     <y>450</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerICPRegistrationModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>outputMRMLNodeComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>298</x>
     <y>291</y>
    </hint>
    <hint type="destinationlabel">
     <x>351</x>
     <y>486</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>registrationButton</sender>
   <signal>clicked()</signal>
   <receiver>qSlicerICPRegistrationModuleWidget</receiver>
   <slot>onModelRegistration()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>284</x>
     <y>541</y>
    </hint>
    <hint type="destinationlabel">
     <x>101</x>
     <y>565</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onModelRegistration()</slot>
 </slots>
</ui>
